containers/src/Data/Map/Internal.hs

```diff
@@ -4118,6 +4119,30 @@ deleteFindMax t = case maxViewWithKey t of
   Nothing -> (error "Map.deleteFindMax: can not return the maximal element of an empty map", Tip)
   Just res -> res
 
+{--------------------------------------------------------------------
+  Iterator
+--------------------------------------------------------------------}
+
+newtype Iterator k a = Iterator (Stack k a)
```

Comment:
As far as I can tell, this `newtype` just introduces noise and makes it less obvious what's being passed around. Why not just use `Stack` directly? If we were using linear types, I could imagine that as a useful public API, but I'm not sure it makes sense in the current context.

Reply:
I can change it to just use `Stack`, don't have a strong opinion on it.

I'm not sure what you mean about linear types though. We could expose this in the public API as it is, if we wanted to.

Reply:
Oh, I meant if we wanted to leave open the option of representing it with a mutable array without a state thread type parameter.

Reply:
Ah I see. Well, removed the newtype for now.